main.go

```diff
@@ -558,6 +558,7 @@ func mainCore() error {
 	webMux.Get("/parameters", explore.ParametersPage)
 	webMux.With(explore.BlockHashPathOrIndexCtx).Get("/block/{blockhash}", explore.Block)
 	webMux.With(explorer.TransactionHashCtx).Get("/tx/{txid}", explore.TxPage)
+	webMux.With(explorer.TransactionHashCtx,explorer.TransactionIoIndexCtx).Get("/tx/{txid}/{inout}/{inoutid}", explore.TxPage)
```

Comment:
To avoid the parsing and checking of {inout}, you could have two routes,

```go
webMux.With(explorer.TransactionHashCtx, explorer.TransactionIndexCtx, explorer.TxnIsIn).Get("/tx/{txid}/in/{inoutid}", explore.TxPage)
webMux.With(explorer.TransactionHashCtx, explorer.TransactionIndexCtx, explorer.TxnIsOut).Get("/tx/{txid}/out/{inoutid}", explore.TxPage)
```

where `explorer.TransactionIndexCtx` only checks `{inoutid}`, `explorer.TxnIsOut` sets `ctxTxInOut` to `"in"`, and  `explorer.TxnIsIn` sets `ctxTxInOut` to `"out"`.

Any path along these lines with anything other than "in" or "out" in that position would not match a handler.

Reply:
But I'm fine with the way you have it as long as we be sure to handle incorrect paths.

Reply:
Ok, i put in the check to validate the in/out/''.  I didn't do anything with the index value at the end.  right now it can be anything still.  you ok with that?  
```
...6eb590adfaf6/in/apples
...6eb590adfaf6/out/bananas
...6eb590adfaf6/in/55334343234

```

These are all valid at present

Reply:
Lol, yeah that's cool.

Reply:
Given the explicit url matching you want it might be best to just use separate routes as you recommend.  Seems like a cleaner solution.  I'm happy to do that later today or tomorrow.  Up to you.